tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -197,14 +198,31 @@ func (a *AwsProvider) CreateWindowsVM() (credentials *types.Credentials, err err
 	// Build new credentials structure to be used by other actors. The actor is responsible for checking if
 	// the credentials are being generated properly. This method won't guarantee the existence of credentials
 	// if the VM is spun up
-	credentials = types.NewCredentials(instanceID, publicIPAddress, decryptedPassword, winUser)
+	credentials := types.NewCredentials(instanceID, publicIPAddress, decryptedPassword, winUser)
+	w.Credentials = credentials
+
+	// Setup Winrm and SSH client so that we can interact with the Windows Object we created
+	if err := w.SetupWinRMClient(); err != nil {
+		return nil, fmt.Errorf("failed to setup winRM client for the Windows VM: %v", err)
+	}
+	// Wait for some time before starting configuring of ssh server. This is to let sshd service be available
+	// in the list of services
+	// TODO: Parse the output of the `Get-Service sshd, ssh-agent` on the Windows node to check if the windows nodes
+	// has those services present
+	time.Sleep(time.Minute)
+	if err := w.ConfigureOpenSSHServer(); err != nil {
+		return w, fmt.Errorf("failed to configure OpenSSHServer on the Windows VM: %v", err)
+	}
+	if err := w.GetSSHClient(); err != nil {
+		return w, fmt.Errorf("failed to get ssh client for the Windows VM created: %v", err)
+	}
+
```

Comment:
This function is getting large and unwieldy. Please move this to a helper.

Reply:
How about we make the helper public and keeps these other methods as private? Is there a reason to expose all these methods? That was the code in the internal framework will become smaller too.